doc/how-tos/load-data-into-pachyderm.md

```diff
@@ -0,0 +1,193 @@
+
+   * To start and finish an atomic commit, run:
+
+     ```bash
+     $ pachctl put file <repo>@<branch>:</path/to/file1> -f <file1>
+     ```
+
+   * To start a commit and add data in iterations:
+
+     1. Start a commit:
+
+        ```sh
+        $ pachctl start commit <repo>@<branch>
+        ```
+     1. Put your data:
+
+        ```bash
+        $ pachctl put file <repo>@<branch>:</path/to/file1> -f <file1>
+        ```
+
+     1. Work on your changes, and when ready, put more data:
+
+        ```bash
+        $ pachctl put file <repo>@<branch>:</path/to/file2> -f <file2>
+        ```
+
+     1. Close the commit:
+
+        ```bash
+        $ pachctl finish commit <repo>@<branch>
+        ```
+
+## Filepath Format
+
+In Pachyderm, you specify the path to file by using the `-f` option. A path
+to file can be a local path or a URL to an external resource. You can add
+multiple files or directories by using the `-i` option. To add contents
+of a directory, use the `-r` flag.
+
+The following table provides examples of `pachctl put file` commands with
+various filepaths and data sources:
+
+* Put data from a URL:
+
+  ```
+  $ pachctl put file <repo>@<branch>:</path/to/file> -f http://url_path
+  ```
+
+* Put data from an object store. You can use `s3://`, `gcs://`, or `as://`
+in your filepath:
+
+  ```
+  $ pachctl put file <repo>@<branch>:</path/to/file> -f s3://object_store_url
+  ```
+
+  **Note:** If you are configuring a local cluster to access an S3 bucket,
+  you need to first deploy a Kubernetes `Secret` for the selected object
+  store.
+
+* Add multiple files at once by using the `-i` option or multiple `-f` flags. In
+the case of `-i`, the target file must be a list of files, paths, or URLs
+that you want to input all at once:
+
+  ```sh
+  $ pachctl put file <repo>@<branch> -i <file containing list of files, paths, or URLs>
+  ```
+
+* Input data from stdin into a data repository by using a pipe:
+
+  ```sh
+  $ echo "data" | pachctl put file <repo>@<branch> -f </path/to/file>
+  ```
+
+* Add an entire directory or all of the contents at a particular URL, either
+HTTP(S) or object store URL, `s3://`, `gcs://`, and `as://`, by using the
+recursive flag, `-r`:
+
+  ```sh
+  $ pachctl put file <repo>@<branch> -r -f <dir>
+  ```
+
+## Loading Your Data Partially
+
+Depending on your use case, you might decide not to import all of your
+data into Pachyderm but only store and apply version control to some
+of it. For example, if you have a 10 PB dataset, loading the
+whole dataset into Pachyderm is a costly operation that takes
+a lot of time and resources. To optimize performance and the
+use of resources, you might decide to load some of this data into
+Pachyderm, leaving the rest of it in its original source.
+
+One possible way of doing this is adding a `.txt` file with a
+URL to the specific file or directory in your dataset to a Pachyderm
+repository and refer to that text file in your pipeline. Pachyderm
```

Comment:
```suggestion
repository and refer to that file in your pipeline. Your pipeline code would read the URL or path in the external data source and retrieve that data as needed for processing instead of needing to preload it all into a Pachyderm repo. This works particularly well for mostly-immutable data as Pachyderm
```